### PR TITLE
findup: Test and remember to use resolved paths 

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [ ] Documentation update

#### Overview of change:

Fixes a bug where findup wouldn't look above the cwd.

### CODE REVIEW

File: src/configuration.ts [hunk 1]
```diff
@@ -159,18 +160,23 @@ function findup(filename: string, directory: string): string | undefined {
...
         // Try reading in the entire directory and looking for a file with different casing.
         const filenameLower = filename.toLowerCase();
-        const result = fs.readdirSync(cwd).find((entry) => entry.toLowerCase() === filenameLower);
-        if (result !== undefined) {
-            showWarningOnce(`Using mixed case tslint.json is deprecated. Found: ` + path.join(cwd, result));
+        const oddlyCasedFile = fs.readdirSync(cwd).find((entry) => entry.toLowerCase() === filenameLower);
+        if (oddlyCasedFile !== undefined) {
+            const fullOddlyCased = path.join(cwd, oddlyCasedFile);
+            showWarningOnce(dedent`
+                Case-insensitive matching will be deprecated in tslint 6.0.
```
Comment (ajafff) [inline src/configuration.ts]: s/deprecated/removed/

**Comment by adidahiya** [comment]
failing CI on windows

**Comment by giladgray** [comment]
@andy-hanson can you please update this PR? merge conflicts/

**Comment by johnwiseheart** [comment]
Closing due to age and inactivity. Feel free to re-open or create a new pull request if you decide to continue working on this.

**Pull request closed without merging**